### PR TITLE
Fix transformers

### LINKED ISSUE #3217: BUG: Zero shap values w/ llama-2 models
### Issue Description

Hello,

I am trying to use shap with a simple HuggingFace text-classification pipeline using one of the llama-2 models; however, the output values from the explainer are all zero. My environment works fine with the multi-class example in the docs w/ BERT, so I believe it is llama-2 specific. 

```
explainer = shap.Explainer(pipeline)
shap_values = explainer(["hello, world!"])

print(shap_values)

.values =
array([[[0., 0.],
        [0., 0.],
        [0., 0.],
        [0., 0.],
        [0., 0.]]])

.base_values =
array([[0.68299288, 0.31700709]])
```

### Minimal Reproducible Example

```python
from transformers import AutoTokenizer, AutoModelForSequenceClassification

model = AutoModelForSequenceClassification.from_pretrained("NousResearch/Llama-2-7b-hf")

tokenizer = AutoTokenizer.from_pretrained("NousResearch/Llama-2-7b-hf")

from transformers import pipeline

pipeline = pipeline(
    "text-classification", model=model, tokenizer=tokenizer, return_all_scores=True
)

import shap

explainer = shap.Explainer(pipeline)
shap_values = explainer(["hello, world!"])

print(shap_values)
```

### Traceback

_No response_

### Expected Behavior

_No response_

### Bug report checklist

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest release](https://github.com/shap/shap/releases) of shap.
- [X] I have confirmed this bug exists on the [master branch](https://github.com/shap/shap/blob/master/CONTRIBUTING.md#installing-from-the-master-branch) of shap.
- [ ] I'd be interested in making a PR to fix this bug

### Installed Versions

Currently using master branch

### PR DESCRIPTION
**Pull request by costrau**
## Overview

Closes #3217

There were two problems when using a current transformer model. The first was fixed in #3536 and resulted in all SHAP values being zero. The second problem is that the list of transformer models is hard-coded, so each new model would require a new entry. Fortunately, the transformers library provides such a list, so I changed the code to check against their lists. 

Minimal working example (random model, random question)
```python
import transformers
import shap

pmodel = transformers.pipeline("text-classification", "microsoft/phi-2", trust_remote_code=True)

data = [
    "Alice: Who is the current president of the United States?\nBob:"
]

pmodel.model.config.task_specific_params = dict({ "text-generation":{
    "max_new_tokens":20}})

pmodel.model.config.is_decoder = True
exp = shap.Explainer(pmodel, pmodel.tokenizer)
shap_values = exp(data, max_evals=15)
shap.plots.text(shap_values)
```

## Checklist

- [x] All [pre-commit checks](https://pre-commit.com/#install) pass.
- [ ] Unit tests added (if fixing a bug or adding a new feature)

### CODE REVIEW

**Comment by costrau** [comment]
The failed test looks like a connection problem to me. Can you rerun the pipeline?

**Comment by CloseChoice** [comment]
> The failed test looks like a connection problem to me. Can you rerun the pipeline?

cc @connortann

**Review by CloseChoice — COMMENTED** [review]
Thanks for the PR. I have not checked this thoroughly yet but this looks amazing. Will come back later this week with a more in-depth review.

**Review by connortann — APPROVED** [review]
Thanks for the PR, it looks great! Apologies for the delay in getting this reviewed & merged.

I pushed a few commits to fixup some merge conflicts (since we introduced the ruff autoformatter), and added a better exception message for the case where the model is not found. Previously it looks like this would have led to an error as the `logits` variable was referenced before being assigned.

PS: it looks like our pre-commit check is failing due to a merge timeout, but not to worry: I've confirmed locally that all pre-commit checks pass.
